src/useTowerGame.ts

```diff
@@ -33,24 +40,115 @@ export const useTowerGame = () => {
+            const directionVariableMap: {
+                [key in DIRECTION]: {
+                    lengthValues: Box3[];
+                    widthValues: Box3[];
+                    missedWidthValues: Box3[];
+                    startValues: Box3[];
+                };
+            } = {
+                [DIRECTION.POSITIVE_X]: {
+                    lengthValues: prevPrev,
+                    widthValues: prevStack,
+                    startValues: stackPrev,
+                    missedWidthValues: stackPrev,
+                },
+                [DIRECTION.NEGATIVE_X]: {
+                    lengthValues: prevPrev,
+                    widthValues: stackPrev,
+                    startValues: prevPrev,
+                    missedWidthValues: prevStack,
+                },
+                [DIRECTION.POSITIVE_Z]: {
+                    lengthValues: prevStack,
+                    widthValues: prevPrev,
+                    missedWidthValues: prevPrev,
+                    startValues: prevStack,
+                },
+                [DIRECTION.NEGATIVE_Z]: {
+                    lengthValues: stackPrev,
+                    widthValues: prevStack,
+                    missedWidthValues: prevPrev,
+                    startValues: stackPrev,
+                },
+                [DIRECTION.NONE]: {
+                    lengthValues: prevPrev,
+                    widthValues: prevPrev,
+                    startValues: prevPrev,
+                    missedWidthValues: prevPrev,
+                },
+            };
 
+            const length =
+                directionVariableMap[overlappedDirection].lengthValues[0].max.z -
+                directionVariableMap[overlappedDirection].lengthValues[1].min.z;
+
+            const width =
+                directionVariableMap[overlappedDirection].widthValues[0].max.x -
+                directionVariableMap[overlappedDirection].widthValues[1].min.x;
+
+            const startX =
+                directionVariableMap[overlappedDirection].startValues[0].min.x + width / 2;
+
+            const startZ =
+                directionVariableMap[overlappedDirection].startValues[1].min.z + length / 2;
+
+            const missedWidth = stackBoxEdges.max.x - previousBoxEdges.max.x;
+            const missedlength = stackBoxEdges.max.z - stackBoxEdges.min.z;
+            const missedStartZ = previousBoxEdges.min.z + missedlength / 2;
+            const missedStartX = previousBoxEdges.max.x + missedWidth / 2;
+
+            return {
+                width,
+                length,
+                startX,
+                startZ,
+                missedWidth,
+            };
+        }
+        const newWidth = lastBoxEdges.max.x - lastBoxEdges.min.x;
+
+        let missedStartX = lastBoxEdges.min.x + newWidth / 2;
+        let missedStartZ = 0;
+        let missedWidth = 0;
+        let missedlength = 0;
+        let directionOverlapped = DIRECTION.NONE;
```

Comment:
Potentially could create a function here that gets the directionOverlapped and pivot off that